packages/bvaughn-architecture-demo/components/sources/utils/updateStringWithExpression.test.ts

```diff
@@ -0,0 +1,61 @@
+import updateStringWithExpression from "./updateStringWithExpression";
+
+// Converts a string like: "foo |bar"
+// Into the string "foo bar" and the cursor index 4
+function getExpressionHelper(text: string, expression: string): string | null {
+  let cursorIndex = text.length;
+
+  for (let index = 0; index < text.length; index++) {
+    const character = text.charAt(index);
+    if (character === "|") {
```

Comment:
:question: where's the `'|'` coming from in the first place? Is that being auto-inserted at the cursor position by Lezer? Are these utils inserting it?  What happens if there's a literal pipe character being written?

Reply:
There is no "|" in the real code. This is just a test helper I wrote so that I can write human-readable tests (with a visible "cursor").